### PR TITLE
New-DbaAgentJob - better performance

### PR DESCRIPTION
**Pull request by andreasjordan**
see https://github.com/dataplat/dbatools/pull/8233#discussion_r825889017

### CODE REVIEW

**Comment by potatoqualitee** [comment]
note: updated branch with new start-dbaencryption

File: functions/New-DbaAgentJob.ps1 [hunk 1]
```diff
@@ -409,7 +409,7 @@ function New-DbaAgentJob {
 
             Add-TeppCacheItem -SqlInstance $server -Type job -Name $Job
 
-            Get-DbaAgentJob -SqlInstance $instance | Where-Object Name -eq $Job
+            Get-DbaAgentJob -SqlInstance $server -Job $Job
```
Comment (lowlydba) [inline functions/New-DbaAgentJob.ps1:412]:
```suggestion
            Get-DbaAgentJob -SqlInstance $server -SqlCredential $SqlCredential -Job $Job
```

Also need to pass creds 😨

[the remark this answers is not in the document]
Comment (lowlydba) [reply functions/New-DbaAgentJob.ps1:412]: Yup, definitely realized that after the fact 😭

**Comment by andreasjordan** [comment]
I think there are some SMO issues...

File: functions/Set-DbaAgentJob.ps1 [hunk 2]
```diff
@@ -510,7 +510,11 @@ function Set-DbaAgentJob {
                 } catch {
                     Stop-Function -Message "Something went wrong changing the job" -ErrorRecord $_ -Target $instance -Continue
                 }
-                Get-DbaAgentJob -SqlInstance $server | Where-Object Name -eq $currentjob.name
+
+                # Refresh the SMO - another bug in SMO? As this should not be needed...
+                $currentjob.Refresh()
```
Comment (potatoqualitee) [inline functions/Set-DbaAgentJob.ps1:515]: I see this a lot, not so sure it's unintended behavior 🤔

**Comment by potatoqualitee** [comment]
Nice work, everyone! 💯

**Pull request merged**